Makefile

```diff
@@ -1,13 +1,22 @@
 .PHONY: clean all test assets dev proto download-crd-deps
-SOURCE_VERSION := v0.13.2
-KUSTOMIZE_VERSION := v0.12.2
-HELM_CRD_VERSION := v0.10.1
+SOURCE_CONTROLLER_VERSION := v0.15.4
+KUSTOMIZE_CONTROLLER_VERSION := v0.14.0
+HELM_CONTROLLER_VERSION := v0.11.2
 
-all: test build
+all: dist/index.html fmt vet test build
```

Comment:
We did that in Flux CLI. Go Fmt is part of the make file. Should we do this @stefanprodan?

Reply:
I agree with Jordan here; I think it's OK to have phony targets that rewrite files, but the main build shouldn't create (more) diffs. If it does, it encourages commits which contain code changes then some random reformatting elsewhere.

This can make it difficult to reconcile with builds in which you commit generated files. For that, you can make people explicitly regenerate the file, and have a guard in the tests (and CI) to make sure the committed, generated file is correct.

(Also see `go build -mod=readonly`.)

Reply:
(You can call me a hypocrite, since all the flux controllers inherit a Makefile structure which does `fmt` in the build, from kubebuilder. I don't notice because I've told emacs to run `go fmt` when I save a file.)